File: algorithm.py

# Made on Arch Linux, BTW.
# mc_modder, 2025

import math
import time

def algorithm(num):
    if num <= 1:
        return 'n'
    if num == 2:
        return 'y'
    if num % 2 == 0:
        return 'n'
    
    
    for i in range(3, int(math.sqrt(num)) + 1, 2):
        if num % i == 0:
            return 'n'
    return 'y'

def calc(starting_num, iterations):
    for i in range(1, iterations):
        print(f"{starting_num} {algorithm(starting_num)}")
        starting_num += 1

def main():
    
    x = int(input("starting number: "))
    y = int(input("iterations: "))
    
    
    start_time = time.perf_counter()
    
    
    calc(x, y)
    print("Done")
    
    
    duration = time.perf_counter() - start_time
    print(f"Calculation time: {duration:.8f} seconds")
    
    
    z = input("Write prime nums? (y/n): ")
    if z.lower() == 'y':
        with open('prime_numbers.txt', 'w') as f:
            for num in range(x, y + 1):
                if algorithm(num) == 'y':
                    f.write(str(num) + '\n')
        print("Done")
    elif z.lower() == 'n':
        print("Finished")
    else:
        print("Program Complete")

if __name__ == "__main__":
    main()

File: algorithm.cpp
#include <iostream>
#include <cmath>
#include <fstream>
#include <chrono>
#include <iomanip>

// Made on Arch Linux, BTW.
// mc_modder, 2025

using namespace std;

char algorithm(int num) {
    if (num <= 1) return 'n';
    if (num == 2) return 'y';
    if (num % 2 == 0) return 'n';
    
    
    for (int i = 3; i <= sqrt(num); i += 2) {
        if (num % i == 0) return 'n';
    }
    return 'y';
}

void calc(int starting_num, int iterations) {
    for (int i = 1; i < iterations; i++) {
        cout << starting_num << " " << algorithm(starting_num) << endl;
        starting_num++;
    }
}

int main() {
    int x, y;
    cout << "starting number: ";
    cin >> x;
    cout << "ending number: ";
    cin >> y;
    
    
    auto start_time = chrono::high_resolution_clock::now();
    
    calc(x, y);
    cout << "Done" << endl;
    
    
    auto end_time = chrono::high_resolution_clock::now();
    auto duration = chrono::duration_cast<chrono::nanoseconds>(end_time - start_time);
    double seconds = static_cast<double>(duration.count()) / 1e9;
    
    cout << "Calculation time: " << fixed << setprecision(8) << seconds << " seconds" << endl;
    
    char z;
    cout << "Write prime nums? (y/n): ";
    cin >> z;
    if (z == 'y' || z == 'Y') {
        ofstream f("prime_numbers.txt");
        for (int num = x; num <= y + 1; num++) {
            if (algorithm(num) == 'y') {
                f << num << '\n';
            }
        }
        cout << "Done" << endl;
    }
    else if (z == 'n' || z == 'N') {
        cout << "Finished" << endl;
    }
    else {
        cout << "Program Finished" << endl;
    }
    return 0;
}
